yb-voyager/src/queryparser/helpers.go

```diff
@@ -334,3 +307,46 @@ func getOneofActiveField(msg protoreflect.Message, oneofName string) protoreflec
 	// Determine which field within the oneof is set
 	return msg.WhichOneof(oneofDescriptor)
 }
+
+// == Generic helper functions ==
+
+// GetStringField retrieves a string field from a message.
+// Sample example:: {column_ref:{fields:{string:{sval:"s"}}  fields:{string:{sval:"tableoid"}}  location:7}
+func GetStringField(msg protoreflect.Message, fieldName string) string {
+	field := msg.Descriptor().Fields().ByName(protoreflect.Name(fieldName))
+	if field != nil && msg.Has(field) {
+		return msg.Get(field).String()
+	}
+	return ""
+}
+
+// GetMessageField retrieves a message field from a message.
+func GetMessageField(msg protoreflect.Message, fieldName string) protoreflect.Message {
+	field := msg.Descriptor().Fields().ByName(protoreflect.Name(fieldName))
+	if field != nil && msg.Has(field) {
+		return msg.Get(field).Message()
+	}
+	return nil
+}
+
+// GetListField retrieves a list field from a message.
+func GetListField(msg protoreflect.Message, fieldName string) protoreflect.List {
+	field := msg.Descriptor().Fields().ByName(protoreflect.Name(fieldName))
+	if field != nil && msg.Has(field) {
+		return msg.Get(field).List()
+	}
+	return nil
+}
+
+// GetQualifiedName extracts the schema and object name from a list.
+func GetQualifiedName(nameList protoreflect.List) (string, string) {
```

Comment:
nit: Name is not clear as per the return type, should be something `GetObjectAndSchemaName`